Formula/cmdg.rb

```diff
@@ -0,0 +1,24 @@
+class Cmdg < Formula
+  desc "Gmail CLI - Copyright Thomas Habets thomas@habets.se 2019"
+  homepage "https://github.com/ThomasHabets/cmdg"
+  url "https://github.com/ThomasHabets/cmdg/archive/cmdg-1.00.tar.gz"
+  sha256 "4a723a812eae27fac1334e14cfc9b92955e6a3c1c43f1de87aa294d72bb7dc25"
+  depends_on "go"
+  def install
+    ENV["GOPATH"] = buildpath
+    system "go", "get", "github.com/mattn/go-runewidth"
+    system "go", "get", "github.com/pkg/errors"
+    system "go", "get", "github.com/sirupsen/logrus"
+    system "go", "get", "golang.org/x/crypto/ssh/terminal"
+    system "go", "get", "golang.org/x/oauth2"
+    system "go", "get", "golang.org/x/text/encoding"
+    system "go", "get", "google.golang.org/api/gmail/v1"
+    system "go", "get", "github.com/ThomasHabets/cmdg/cmd/cmdg"
+    system "go", "build", "github.com/ThomasHabets/cmdg/cmd/cmdg"
+    bin.install "./cmdg" => "cmdg"
+  end
+
+  test do
+    system "#{bin}/cmdg", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.